src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/SortOrderQueryBuilder.java

```diff
@@ -12,23 +21,39 @@
     private static final String SPACE = " ";
     private static final String COMMA = ",";
 
+    private static final Set<String> VALID_COLUMNS_FOR_ORDER_BY
+        = Collections.unmodifiableSet((Set<? extends String>) Stream
+                                          .of(CaseDetailsEntity.CREATED_DATE_FIELD_COL,
+                                              CaseDetailsEntity.LAST_MODIFIED_FIELD_COL,
+                                              CaseDetailsEntity.JURISDICTION_FIELD_COL,
+                                              CaseDetailsEntity.CASE_TYPE_ID_FIELD_COL,
+                                              CaseDetailsEntity.STATE_FIELD_COL,
+                                              CaseDetailsEntity.REFERENCE_FIELD_COL,
+                                              CaseDetailsEntity.SECURITY_CLASSIFICATION_FIELD_COL)
+                                          .collect(Collectors.toCollection(HashSet::new)));
+
     public String buildSortOrderClause(MetaData metaData) {
         StringBuilder sb = new StringBuilder();
         metaData.getSortOrderFields().forEach(sortOrderField -> {
-            if (sortOrderField.isMetadata()) {
+            if (sortOrderField.isMetadata() && VALID_COLUMNS_FOR_ORDER_BY.contains(getMataFieldName(sortOrderField.getCaseFieldId()))) {
                 sb.append(getMataFieldName(sortOrderField.getCaseFieldId()));
-            } else {
+                appendOrder(sb, sortOrderField);
+            } else if (sortOrderField.getCaseFieldId().matches(VALID_FIELD_NAME_REGEX)) {
                 sb.append(convertFieldNameToJSONBsqlFormat(sortOrderField.getCaseFieldId()));
+                appendOrder(sb, sortOrderField);
             }
-            sb.append(SPACE);
-            sb.append(sortOrderField.getDirection());
-            sb.append(COMMA);
-            sb.append(SPACE);
         });
         // always sort with creation_date as a last order so that it supports cases where no values at all for the configured fields and also default fallback.
         return sb.append(CREATED_DATE + SPACE + SortDirection.fromOptionalString(metaData.getSortDirection())).toString();
     }
 
+    private void appendOrder(final StringBuilder sb, final SortOrderField sortOrderField) {
+        sb.append(SPACE);
+        sb.append(sortOrderField.getDirection());
```

Comment:
I would put the validation here, at the exact point where the data is being converted to SQL, and throw an exception if the sort order is invalid. Otherwise I have to look around a bunch of different places to figure out if the validation always happens.

Reply:
Not sure I understand - the appending happens in the if else as well?

Reply:
ok I will add a top level if block looking for special characters:

`       metaData.getSortOrderFields().forEach(sortOrderField -> {
            if (!sortOrderField.getCaseFieldId().matches(SPECIAL_CHARS)) {
                if (sortOrderField.isMetadata()) {
                    sb.append(getMataFieldName(sortOrderField.getCaseFieldId()));
                } else {
                    sb.append(convertFieldNameToJSONBsqlFormat(sortOrderField.getCaseFieldId()));
                }
                sb.append(SPACE);
                sb.append(sortOrderField.getDirection());
                sb.append(COMMA);
                sb.append(SPACE);
            }
        });`